### ISSUE TITLE
ng-show with translate-values 

### ISSUE THREAD

**Issue by charlie-ablett** [angular-translate/angular-translate#1854]
### Subject of the issue
Sorry if this has already been shown, but I can't find it anywhere. 

I'm using directives (`ng-show`, `ng-hide`, `ng-if`, etc) in translate strings, and it's not working for me. I'm using `translate-compile`.

### Your environment
* version of angular-translate
2.15.2
* version of angular
1.5.3
* which browser and its version
Chromium | 68.0.3440.75 (Official Build) (64-bit)

### Steps to reproduce

Run the [plunkr](https://plnkr.co/edit/Pacl7EXx3dgVrGX2rDYj?p=preview)

### Expected behaviour
I should see "Variable is 1.2.3." 

### Actual behaviour
Even when `variable` is set, `Variable is missing!` appears

**Comment by knalli** [comment]
Hello,

as we had already mentioned in the [issue's template](https://github.com/angular-translate/angular-translate/blob/master/.github/ISSUE_TEMPLATE.md), we need a working and as minimal as possible demo covering your specific use case or issue. 

Follow these steps:
1. Please open [this plnkr base](https://plnkr.co/edit/8vRLnfGmtg51qZFfDeVc?p=preview).
2. Fork it (at the top left).
3. Ensure the version of AngularJS and angular-translate is correct.
4. Ensure all angular-translate plugins are available; reduce and remove anything you can.
5. Write a minimal as possible demo for your specific issue.
6. Save and freeze. Provide us the final link to your demo.

**Comment by knalli** [comment]
Note/reminder:

It looks like your environment is using an old or older release of AngularJS and/or angular-translate. After updating your dependencies, please check and verify your issue again.

**Comment by charlie-ablett** [comment]
The template was different (and had heaps of console errors) when I checked earlier. Hence why I didn't submit a code sample. I'll try your new one now.

**Comment by charlie-ablett** [comment]
Question updated as requested. 
Other than this little issue, the library is solid. Thank you for all your work :)

**Comment by charlie-ablett** [comment]
@knalli Thanks for your attention so far. On my project and on the plunkr above (https://plnkr.co/edit/Pacl7EXx3dgVrGX2rDYj?p=preview) I've updated the version of angular-translate, angular-sanitize and angularjs to latest. The issue persists - would you be able to take look again?

Many thanks.

**Comment by knalli** [comment]
@charlie-ablett You have to escape the interpolation `'<span ng-show="variable">…'` => `'<span ng-show="\'{{variable}}\'">…'`.

See also working demo: https://plnkr.co/edit/64C3QrKJJGZ9KKhoRvZs?p=preview

The reason for this: `translate-compile` only compiles the translation which means basically you have access to the actual current context (scope). Said this, `<span ng-show="app.someVariable">` would work, however, that would probably not be useful.

The `translate-values` aren't the scope of the compilation interpolation, but of the translation interpolation (one step before).

In a nutshell: The interpolation step must render the result (which means it will not work with complex structures because of serialization) and the result of this must be escaped probably for AngularJS.

```
variable = 'test'

// translation interpolation of '<span ng-show="\'{{variable}}\'">'
<span ng-show="\'test\'">

// AJS template interpolation of the string above (no access to the values any more)
<span ng-show=truthy>
```
